controllers/storagecluster/topology.go

```diff
@@ -280,3 +293,83 @@ func (r *StorageClusterReconciler) reconcileNodeTopologyMap(sc *ocsv1.StorageClu
 
 	return nil
 }
+
+// nodesHaveIdenticalValuesForKeys will return true only if
+// a. all the nodes have all the keys in their map
+// b. for a given node, values for all the keys are the same
+func nodesHaveIdenticalValuesForKeys(nodes *corev1.NodeList, keys []string) bool {
+
+	var match = true
+	if len(keys) == 0 {
+		return match
+	}
+
+	for _, node := range nodes.Items {
+		v, found := node.Labels[keys[0]]
+		if !found {
+			match = false
+		}
+		for _, key := range keys {
+			vnext, found := node.Labels[key]
+			if !found {
+				match = false
+			}
+			if v != vnext {
+				match = false
+			}
+		}
+	}
+	return match
+}
+
+func filterDuplicateLabels(sc *ocsv1.StorageCluster, nodes *corev1.NodeList, topologyMap *ocsv1.NodeTopologyMap) {
+
+	if "" == getFailureDomain(sc) {
+		//This is the first time we are determining the failure domain
+
+		if topologyMap.ContainsKey(corev1.LabelZoneFailureDomainStable) {
+			// New label is found, use it and discard old
+			delete(topologyMap.Labels, corev1.LabelZoneFailureDomain)
+			delete(topologyMap.Labels, "failure-domain.kubernetes.io/zone")
```

Comment:
Throughout this function, would it make sense to use the consts you defined in reconcile.go?

Reply:
Done